src/abstractitensornetwork.jl

```diff
@@ -870,6 +870,59 @@ function ITensors.commoninds(tn1::AbstractITensorNetwork, tn2::AbstractITensorNe
   return inds
 end
 
+"""Add two itensornetworks together by growing the bond dimension. The network structures need to be have the same vertex names, same site index on each vertex """
+function add(tn1::AbstractITensorNetwork, tn2::AbstractITensorNetwork)
+  @assert issetequal(vertices(tn1), vertices(tn2))
+
+  if !issetequal(edges(tn1), edges(tn2))
+    new_edges = union(edges(tn1), edges(tn2))
+    tn1 = insert_missing_internal_inds(tn1, new_edges)
+    tn2 = insert_missing_internal_inds(tn2, new_edges)
+    return add(tn1, tn2)
```

Comment:
I would remove this line and let it continue executing the rest of `add`, so it doesn't do these checks again and avoids any potential infinite loops.